src/ig/IGExporter.ts

```diff
@@ -133,61 +125,54 @@ export class IGExporter {
           generation: 'html',
           page: [] // index.[md|html] is required and added later
         },
-        // Parameters apparently required by IG Publisher (as of Jan 29, 2020)
-        parameter: [
-          {
-            code: 'copyrightyear',
-            value: '' // Gets set when ig.ini is processed
-          },
-          {
-            code: 'releaselabel',
-            value: '' // Gets set when ig.ini is processed
-          },
-          {
-            code: 'show-inherited-invariants',
-            value: 'false' // TODO: Make this configurable
-          }
-        ]
+        // required parameters are enforced in the configuration
+        // default to empty array in case we want to add other parameters later
+        parameter: this.config.parameters ?? []
       }
     };
-
     // Add the path-history, if applicable (only applies to HL7 IGs)
-    if (/^https?:\/\/hl7.org\//.test(this.pkg.packageJSON.canonical)) {
+    if (
+      /^https?:\/\/hl7.org\//.test(this.config.canonical) &&
+      !this.ig.definition.parameter.some(param => {
+        return param.code === 'path-history';
+      })
+    ) {
       this.ig.definition.parameter.push({
         code: 'path-history',
-        value: `${this.pkg.packageJSON.canonical}/history.html`
+        value: `${this.config.canonical}/history.html`
       });
     }
-
-    // Add the dependencies
-    if (this.pkg.packageJSON.dependencies) {
+    // check for some optional properties and add them if present
+    if (this.config.useContext?.length) {
+      this.ig.useContext = this.config.useContext;
+    }
```

Comment:
Why can't `useContext` be added the same was as the other properties that are being copied from `this.config` to `this.ig`? What is unique about it in this case?